### PR TITLE
[9.0][IMP] purchase_request: add cancelled state to lines.

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
related to https://github.com/OCA/purchase-workflow/issues/396.

A new state is added to lines to enhance the behavior and flow of purchase_request and its submodules.

### CODE REVIEW

**Comment by LoisRForgeFlow** [comment]
cc @benwillig @aheficent @jbeficent @sbidoul @gva-acsone

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: I had to uninstall purchase_request_procurement to cancel the procurement and test properly

File: purchase_request/views/purchase_request_view.xml [hunk 1]
```diff
@@ -249,6 +252,7 @@
                     <field name="analytic_account_id"
                            groups="analytic.group_analytic_accounting"/>
                     <field name="supplier_id"/>
+                    <field name="cancelled"/>
```
Comment (benwillig) [inline purchase_request/views/purchase_request_view.xml]: @lreficent Can we make the field "cancelled" invisible in all tree view? The line is already grey when it has been cancelled so I guess we can hide it.

**Review by benwillig — CHANGES_REQUESTED** [review]

File: purchase_request/views/purchase_request_view.xml (hunk 1 above)
Comment (LoisRForgeFlow) [reply purchase_request/views/purchase_request_view.xml]: I'm not sure that the user would realize about why the line is gray, that's why I prefer to show the field but yes, maybe you are right and I have to trust more the user :sweat_smile:

**Review by benwillig — APPROVED** [review]

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM :+1:

**Review by JordiBForgeFlow — APPROVED** [review]

**Review by MiquelRForgeFlow — COMMENTED** [review]
Please add migration scripts :octocat:

**Pull request merged**